[yiyang74] iP

yiyang74 opened the pull request.

yiyang74 committed:
Renamed Duke to Echo

yiyang74 committed:
Name: rename Echo to Baus
Feature: add echoing commands and exit on "bye"

yiyang74 committed:
Add ability to store and display stored text

yiyang74 committed:
Add Task.java
Add mark/unmark features
Include handling of some edge cases

yiyang74 committed:
Rename Chatbot from Baus to Beru
Improved clarity in variable names

yiyang74 committed:
Improve code quality:

Reduce length of methods
Reduce magic numbers
Incorporate elements of SLAP
Ulitise refactoring methods

yiyang74 committed:
Add support for 3 types of tasks:
Todos, Deadlines and Events

yiyang74 committed:
Add default case to parseUserSentence

vegetablestabber commented on src/main/java/Beru.java:
```diff
@@ -10,8 +11,7 @@ private static void printList(Task[] tasks) {
         for (int i = 0; i < Task.getTaskCount(); i++) {
-            System.out.println((i + 1) +
-                    ".[" + tasks[i].getStatusIcon() + "] " + tasks[i].getDescription());
+            System.out.println((i + 1) + "." + tasks[i].toString());
```
`tasks[i].toString()` can just be `tasks[i]` when concatenating it to a String. It is, of course, more explicit to the reader but the reader can also figure out that the String representation is derived implicity from a `Task` instead.

yiyang74 replied:
Thanks for pointing it out! Forgot about this one 😅

vegetablestabber commented on src/main/java/Beru.java:
```diff
@@ -57,6 +57,29 @@ private static void printAlreadyUnmarked() {
+                Task.getTaskCount() + " tasks in the list.\n" + LINE_SEPARATOR);
+    }
+
+    private static void printInvalidDeadline() {
```
I like how you have multiple methods to handle different types of invalid input, but maybe you could also consider throwing an `Exception` and handling it in the function where it is called?

This would allow you to disconnect printing from the error message and delegates the responsibility of handling the error to the method (e.g. `handleDeadlineCommand`) where the method that is throwing the `Exception` (e.g. `printInvalidDeadline`).

I noticed that you only handle errors by printing so shifting the responsibility to the parent method is something else you might find helpful as you can avoid confusion between using `print` or `println` for every error printing method, but I also understand the straightforwardness of your solution.

vegetablestabber commented on src/main/java/Beru.java:
```diff
@@ -91,6 +114,61 @@ private static void handleListCommand(Task[] tasks) {
+    }
+
+    private static void handleEventCommand(Task[] tasks, String[] words) {
+        int indexOfFrom = -1;
```
Maybe you could consider creating a new function for obtaining parameters from a given input? This might be useful for any future commands which require parameters and could also be used to handle missing parameters, thus reducing the size of methods like `handleDeadlineCommand` and delegating these responsibilities away into other modular methods.

yiyang74 replied:
oh good idea!

vegetablestabber commented on src/main/java/Beru.java:
```diff
@@ -91,6 +114,61 @@ private static void handleListCommand(Task[] tasks) {
+        return String.join(" ", words);
+    }
+
     private static void parseUserSentence(String sentence, Task[] tasks) {
```
I like the simplicity of this function. A simple `switch` statement is clear and readable to me as a reviewer.

vegetablestabber reviewed (commented):
Looks great! Liked the modularity of your code and suggested some changes in that regard.

OmJRamdas commented on src/main/java/Beru.java:
```diff
@@ -57,6 +57,29 @@ private static void printAlreadyUnmarked() {
+                "Invalid entry for Deadline!\nPlease include \"/by\"!\n" + LINE_SEPARATOR);
+    }
+
+    private static void printMissingEventFields() {
```
Good Naming conventions, its easy to read

OmJRamdas commented on src/main/java/Beru.java:
```diff
@@ -104,9 +182,14 @@ private static void parseUserSentence(String sentence, Task[] tasks) {
+            handleEventCommand(tasks, words);
             break;
         }
     }
```
Good job in avoiding nesting

OmJRamdas commented on src/main/java/Beru.java:
```diff
@@ -104,9 +182,14 @@ private static void parseUserSentence(String sentence, Task[] tasks) {
         case "unmark":
             handleUnmarkedCommand(tasks, words);
             break;
-        default:
```
Could possibly abstract this from Beru to Task class

yiyang74 replied:
Do you mean handle the incorrect input format in Task?

OmJRamdas commented on src/main/java/Beru.java:
```diff
@@ -28,7 +28,7 @@ private static void printTaskStatus(Task t) {
         }
-        System.out.println("  [" + t.getStatusIcon() + "] " + t.getDescription());
+        System.out.println("  " + t);
         System.out.print(LINE_SEPARATOR);
```
Create a function for printing LINE_SEPERATOR

yiyang74 replied:
oh is that better? I'm not sure

yiyang74 committed:
Made Task class abstract

yiyang74 committed:
Change Chatbot name from Beru to Uncle Roger

yiyang74 committed:
Add exceptions instead of using if blocks
to handle errors

yiyang74 committed:
Merge branch 'branch-Level-5'

yiyang74 committed:
Packages: Add uncleroger, uncleroger.ui,
uncleroger.exception and uncleroger.task

printTaskStatus: Change Uncle Roger dialogue

yiyang74 committed:
Merge branch 'branch-A-Packages'

yiyang74 committed:
Change tasks from an Array to ArrayList

yiyang74 committed:
Add support for deleting tasks from the list

Alfred-Goh02 commented on src/main/java/uncleroger/ui/UncleRoger.java:
```diff
@@ -0,0 +1,300 @@
+import uncleroger.task.*;
+import java.util.Arrays;
+import java.util.Scanner;
+
```
Consider separating command handling methods into another class file, UI methods can also be separated into another class, event though this file is packaged as UI

Alfred-Goh02 commented on src/main/java/uncleroger/ui/UncleRoger.java:
```diff
@@ -0,0 +1,300 @@
+        String[] words = sentence.split(" ");
+        String command = words[0];
+        switch (command.toLowerCase()) {
+        case "list":
```
Command strings such as list, mark etc should be declared as constants, avoid magic literals.

Alfred-Goh02 commented on src/main/java/uncleroger/task/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+package uncleroger.task;
+
```
Good job, all your task classes are very concise

yiyang74 committed:
Add UncleRoger.txt to .gitignore

yiyang74 committed:
Save feature:
create .txt file and directory if not present
load data in .txt file if present
add new list entries to .txt file

yiyang74 committed:
Merge branch 'branch-Level-6'

yiyang74 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/uncleroger/ui/UncleRoger.java

yiyang74 committed:
Add MANIFEST.MF to .gitignore

yiyang74 committed:
Improve save feature

yiyang74 committed:
Make code more OOP

yiyang74 committed:
Add find feature

yiyang74 committed:
Add JavaDoc comments

yiyang74 committed:
Merge pull request #1 from yiyang74/branch-Level-9

Branch level 9

yiyang74 committed:
Merge branch 'master' into branch-A-JavaDoc

yiyang74 committed:
Merge pull request #2 from yiyang74/branch-A-JavaDoc

Branch a java doc

yiyang74 committed:
Add a User Guide

yiyang74 committed:
Add more JavaDocs

yiyang74 committed:
Update README.md

yiyang74 committed:
Add exception when no index is provided for mark/unmark/delete
Modify save feature to save after every command

yiyang74 committed:
Add exceptions for when /by /from /to fields are empty
for deadline and event tasks

yiyang74 committed:
Change variable name
